### ISSUE TITLE
Running on multiple devices is broken 

### ISSUE THREAD

**Issue by mit-mit** [flutter/flutter#14879]
`flutter run -d all` on a new app

=>

```
mit-macbookpro3:platform_adaptation mit$ flutter run -d all
Launching lib/main.dart on Android SDK built for x86 in debug mode...
Initializing gradle...                                1.3s
Resolving dependencies...                             1.9s
Running 'gradlew assembleDebug'...                    2.4s
Built build/app/outputs/apk/app-debug.apk (22.6MB).
Installing build/app/outputs/apk/app.apk...           1.4s
I/FlutterActivityDelegate( 4252): onResume setting current activity to this
Launching lib/main.dart on iPhone 7 in debug mode...
Launching lib/main.dart on iPhone 8 in debug mode...
Error 1001 received from application: File system already exists
Error initializing DevFS: NoSuchMethodError: The method '[]' was called on null.
Receiver: null
Tried calling: []("uri")
mit-macbookpro3:platform_adaptation mit$
```

**Comment by mit-mit** [comment]
Cc @asiva

**Comment by mit-mit** [comment]
This should be resolved now (tested in beta 0.3.1)
